sandbox/cgroup/tracking.go

```diff
@@ -108,11 +114,26 @@ tryAgain:
 	//
 	// Verify the effective tracking cgroup and check that our scope name is
 	// contained therein.
-	path, err := cgroupProcessPathInTrackingCgroup(pid)
-	if err != nil {
-		return err
+	var pathAfter string
+	// XXX: It seems that systemd is racy and can not actually complete the
+	// request after returning from StartTransientUnit. Iterate for up to 500ms
+	// to see systemd perform the move. This is inherently racy.
+	// TODO: After pid-based tracking is removed, remove this spin logic as
+	// synchronicity is no longer relevant. It is only relevant during the
+	// transition period while both snap run and snap-confine manipulate
+	// cgroups.
+	for i := 0; i < 50; i++ {
```

Comment:
Is there something in the API we could wait for instead?

Reply:
No, that's the thing. It feels like a bug as it was entirely synchronous before.

Reply:
I will chase the systemd side of this now, at least to report a bug.